### PR TITLE
Prevent registration outputs from being edited

### LINKED ISSUE #1: Prevent atlas layers etc. from being editable
Currently `layer.editable = False` doesn't work when the layer is not visible (if visibility is toggled, then the layer becomes editable again ([see napari issue](https://github.com/napari/napari/issues/1346)).

Once this is fixed, set atlas layer to not be editable, and [ideally lock all layer properties other than visibility](https://github.com/napari/napari/issues/1570).

### PR DESCRIPTION
**Pull request by adamltyson**
Closes #1 

Leaving the idea of locking the layers for now. We should retain some flexibility for the advanced user (they can make the layers editable via the console if they like).

### CODE REVIEW

File: brainreg_segment/segment.py [hunk 1]
```diff
@@ -414,9 +414,15 @@ def initialise_loaded_data(self):
...
             self.hemispheres_data = self.viewer.layers["Hemispheres"].data
 
+        self.prevent_layer_edit()
+
         self.initialise_segmentation_interface()
         self.status_label.setText("Ready")
 
+    def prevent_layer_edit(self):
+        for layer in self.viewer.layers:
+            layer.editable = False
```
Comment (dstansby) [inline brainreg_segment/segment.py:424]: Will this lock all the layers a user has loaded before loading anything with `brainreg-segment`? If so isn't that quite blunt, and would it be possible to just set `editable` on the layers as they're loaded above?

File: tests/tests/test_integration/test_gui/test_gui.py [hunk 2]
```diff
@@ -51,6 +53,13 @@ def check_loaded_layers(widget, num_layers):
     assert widget.atlas_layer.name == widget.atlas.atlas_name
 
 
+def check_editable(widget, standard_space=False):
```
Comment (dstansby) [inline tests/tests/test_integration/test_gui/test_gui.py:56]: Recommend renaming to `check_not_editable` to better describe what it's checking

**Review by dstansby — CHANGES_REQUESTED** [review]

File: brainreg_segment/segment.py (hunk 1 above)
Comment (adamltyson) [reply brainreg_segment/segment.py:424]: Ah yes good point.

Comment (adamltyson) [reply brainreg_segment/segment.py]: I've addressed this. Only layers added by the widget are set to non-editable. I thought I could just `add_image(data, editable=False)` but that's not supported.

File: brainreg_segment/segment.py [hunk 3]
```diff
@@ -414,11 +425,46 @@ def initialise_loaded_data(self):
...
+                self.hemispheres_string
+            ]
+            self.hemispheres_data = self.hemispheres_layer.data
+
+        self.prevent_layer_edit()
 
         self.initialise_segmentation_interface()
         self.status_label.setText("Ready")
 
+    def collate_widget_layers(self):
```
Comment (dstansby) [inline brainreg_segment/segment.py:438]:
```suggestion
    def collate_widget_layers(self):
    """
    Populate self.editable_widget_layers and self.non_editable_widget_layers.
    """
```

**Comment by dstansby** [comment]
Apart from one small suggestion, and a recommendation for a test (see above), this looks good

**Comment by adamltyson** [comment]
For reasons I don't understand, the tests are passing when pushing to the branch, but the PR tests are failing. Any ideas @dstansby?

File: brainreg_segment/segment.py [hunk 4]
```diff
@@ -464,27 +510,26 @@ def remove_layers(self):
         TODO: This needs work. Runs into an error currently
         when switching from a annotated project to another one
         """
-        if len(self.viewer.layers) != 0:
+        self.collate_widget_layers()
```
[the remark this answers is not in the document]
Comment (adamltyson) [reply brainreg_segment/segment.py:517]: Added something simple in https://github.com/brainglobe/brainreg-segment/pull/75/commits/df05671b6ca906d56b36f81085de5c21857ddbe7

**Comment by adamltyson** [comment]
I added a test, but I don't get why the tests are passing on push, but failing on PR.

**Comment by dstansby** [comment]
Weird, could be because one of them is rebasing on top of the current main branch first before runinng the tests?

**Comment by dstansby** [comment]
Yeah, for some reason the `push` job is checking out the `editable` branch of the repository, whereas the `pull_request` job is checking out this PR. So the test fail is real I think. I've opened https://github.com/brainglobe/brainreg-segment/pull/90 to fix the double CI

**Comment by adamltyson** [comment]
> Weird, could be because one of them is rebasing on top of the current main branch first before runinng the tests?

Makes sense. Think I've fixed it now.

**Comment by adamltyson** [comment]
I've pushed changes, but GitHub seems to be having problems, so they're not showing up here. Leaving this reminder to check that they show up eventually.

File: tests/tests/test_integration/test_gui/test_gui.py [hunk 5]
```diff
@@ -41,6 +42,22 @@ def test_load_standard_space(segmentation_widget):
...
+    before brainreg-segment added layers
+    """
+    assert len(segmentation_widget.viewer.layers) == 0
+    segmentation_widget.viewer.add_points(np.array([[1, 1], [2, 2]]))
+    assert len(segmentation_widget.viewer.layers) == 1
+    segmentation_widget.standard_space = False
+    segmentation_widget.plugin = "brainglobe-napari-io.brainreg_read_dir"
+    segmentation_widget.directory = brainreg_dir
+    segmentation_widget.load_brainreg_directory()
+    check_loaded_layers(segmentation_widget, 8)
```
Comment (dstansby) [inline tests/tests/test_integration/test_gui/test_gui.py:60]:
```suggestion
    # Brainreg should load 7 new layers
    check_loaded_layers(segmentation_widget, 8)
```

**Review by dstansby — APPROVED** [review]
Looks 👍 , I've just suggested a clarifying comment in a test, feel free to take or leave it and self-merge

**Pull request merged**